packages/camera/camera_avfoundation/ios/camera_avfoundation/Sources/camera_avfoundation/CameraPlugin.swift

```diff
@@ -0,0 +1,520 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Flutter
+import ObjectiveC
+
+// Import Objectice-C part of the implementation when SwiftPM is used.
+#if canImport(camera_avfoundation_objc)
+  import camera_avfoundation_objc
+#endif
+
+public typealias CaptureNamedDeviceFactory = (String) -> FLTCaptureDevice
+
+public final class CameraPlugin: NSObject, FlutterPlugin, FCPCameraApi {
+  let registry: FlutterTextureRegistry
+  let messenger: FlutterBinaryMessenger
+  let globalEventAPI: FCPCameraGlobalEventApi
+  let deviceDiscoverer: FLTCameraDeviceDiscovering
+  let permissionManager: FLTCameraPermissionManager
+  let captureDeviceFactory: CaptureNamedDeviceFactory
+  let captureSessionFactory: CaptureSessionFactory
+  let captureDeviceInputFactory: FLTCaptureDeviceInputFactory
+
+  /// All FLTCam's state access and capture session related operations should be on run on this queue.
+  var captureSessionQueue: DispatchQueue
+
+  /// An internal camera object that manages camera's state and performs camera operations.
+  var camera: FLTCam?
+
+  public static func register(with registrar: FlutterPluginRegistrar) {
+    let instance = CameraPlugin(registry: registrar.textures(), messenger: registrar.messenger())
+
+    SetUpFCPCameraApi(registrar.messenger(), instance)
+  }
+
+  convenience init(registry: FlutterTextureRegistry, messenger: FlutterBinaryMessenger) {
+    self.init(
+      registry: registry,
+      messenger: messenger,
+      globalAPI: FCPCameraGlobalEventApi(binaryMessenger: messenger),
+      deviceDiscoverer: FLTDefaultCameraDeviceDiscoverer(),
+      permissionManager: FLTCameraPermissionManager(
+        permissionService: FLTDefaultPermissionService()),
+      deviceFactory: { name in
+        FLTDefaultCaptureDevice(device: AVCaptureDevice(uniqueID: name)!)
```

Comment:
I don't like the null assertion here, but it is consistent with the ObjC implementation behavior, and I don't want to go into the error handling improvements rabbit hole during the migration